### PR TITLE
Cleanup mislabeled BSD license

### PR DESCRIPTION
**Pull request by tfoote**
This was propagated due to an invalid license in ament_lint which is proposed to be fixed at: https://github.com/ament/ament_lint/pull/209

### CODE REVIEW

File: include/rcpputils/filesystem_helper.hpp [hunk 1]
```diff
@@ -1,35 +1,31 @@
...
+//      notice, this list of conditions and the following disclaimer.
+
+//    * Redistributions in binary form must reproduce the above copyright
+//      notice, this list of conditions and the following disclaimer in the
+//      documentation and/or other materials provided with the distribution.
+
+//    * Neither the name of the copyright holder nor the names of its
+//      contributors may be used to endorse or promote products derived from
+//      this software without specific prior written permission.
+
```
Comment (wjwwood) [inline include/rcpputils/filesystem_helper.hpp]: Not having `//` here looks wrong to me.

**Comment by dirk-thomas** [comment]
> maybe @dirk-thomas can comment, since they were added in #25?

#25 is only a redo of #4. And in these patches only the license of the visibility macros is BSD (see https://github.com/ros2/rcpputils/pull/25/files#diff-974160601a417a6b78e21e73844e59f3). I think that file can safely be changed to Apache. I haven't looked at the other cases.

**Comment by emersonknapp** [comment]
@tfoote is the expectation that the lint check here will pass when https://github.com/ament/ament_lint/pull/209 is merged? In what order should things get merged?

**Comment by tfoote** [comment]
This should pass with ament/ament_lint#205 it should land before ament/ament_lint#209 causes it to fail.

File: include/rcpputils/visibility_control.hpp [hunk 2]
```diff
@@ -1,33 +1,29 @@
...
-// FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
-// COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
-// INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
-// BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
-// LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
-// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
-// LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
-// ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// modification, are permitted provided that the following conditions are met:
+
```
Comment (ahcorde) [inline include/rcpputils/visibility_control.hpp]: Missing //

File: include/rcpputils/visibility_control.hpp [hunk 3]
```diff
@@ -1,33 +1,29 @@
...
+//    * Redistributions of source code must retain the above copyright
+//      notice, this list of conditions and the following disclaimer.
+
```
Comment (ahcorde) [inline include/rcpputils/visibility_control.hpp]: Missing //

File: include/rcpputils/visibility_control.hpp [hunk 4]
```diff
@@ -1,33 +1,29 @@
...
+//    * Redistributions in binary form must reproduce the above copyright
+//      notice, this list of conditions and the following disclaimer in the
+//      documentation and/or other materials provided with the distribution.
+
```
Comment (ahcorde) [inline include/rcpputils/visibility_control.hpp]: Missing //

File: include/rcpputils/visibility_control.hpp [hunk 5]
```diff
@@ -1,33 +1,29 @@
...
+//    * Neither the name of the copyright holder nor the names of its
+//      contributors may be used to endorse or promote products derived from
+//      this software without specific prior written permission.
+
```
Comment (ahcorde) [inline include/rcpputils/visibility_control.hpp]: Missing //

**Comment by ahcorde** [comment]
@tfoote  are the changes in the licenses still need it? I think the copyright test is passing in master, right ?

But the change in the `package.xml` is needed

File: include/rcpputils/filesystem_helper.hpp [hunk 6]
```diff
@@ -1,35 +1,31 @@
 // Copyright (c) 2019, Open Source Robotics Foundation, Inc.
-// All rights reserved.
+
```
[the remark this answers is not in the document]
Comment (clalancette) [reply include/rcpputils/filesystem_helper.hpp]: Fixed now.

File: include/rcpputils/split.hpp [hunk 7]
```diff
@@ -1,35 +1,31 @@
...
-// FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
-// COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
-// INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
-// BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
-// LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
-// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
-// LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
-// ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// modification, are permitted provided that the following conditions are met:
+
```
[the remark this answers is not in the document]
Comment (clalancette) [reply include/rcpputils/split.hpp]: Fixed now.

File: include/rcpputils/split.hpp [hunk 8]
```diff
@@ -1,35 +1,31 @@
...
+//    * Redistributions of source code must retain the above copyright
+//      notice, this list of conditions and the following disclaimer.
+
```
[the remark this answers is not in the document]
Comment (clalancette) [reply include/rcpputils/split.hpp]: Fixed now.

File: include/rcpputils/split.hpp [hunk 9]
```diff
@@ -1,35 +1,31 @@
...
+//    * Redistributions in binary form must reproduce the above copyright
+//      notice, this list of conditions and the following disclaimer in the
+//      documentation and/or other materials provided with the distribution.
+
```
[the remark this answers is not in the document]
Comment (clalancette) [reply include/rcpputils/split.hpp]: Fixed now.

File: include/rcpputils/split.hpp [hunk 10]
```diff
@@ -1,35 +1,31 @@
...
+//    * Neither the name of the copyright holder nor the names of its
+//      contributors may be used to endorse or promote products derived from
+//      this software without specific prior written permission.
+
```
[the remark this answers is not in the document]
Comment (clalancette) [reply include/rcpputils/split.hpp]: Fixed.

File: include/rcpputils/visibility_control.hpp (hunk 2 above)
Comment (clalancette) [reply include/rcpputils/visibility_control.hpp]: Fixed.

File: include/rcpputils/visibility_control.hpp (hunk 3 above)
Comment (clalancette) [reply include/rcpputils/visibility_control.hpp]: Fixed.

File: include/rcpputils/visibility_control.hpp (hunk 4 above)
Comment (clalancette) [reply include/rcpputils/visibility_control.hpp]: Fixed.

File: include/rcpputils/visibility_control.hpp (hunk 5 above)
Comment (clalancette) [reply include/rcpputils/visibility_control.hpp]: Fixed.

**Comment by clalancette** [comment]
@ahcorde This change is still needed to update the license from "BSD License 2.0" (which doesn't really exist) to the proper BSD 3-Clause.  The license wording doesn't change, only the formatting and what we call it.  Thus, I think this is still needed.

Another review pass would be appreciated.  Thanks!

**Review by emersonknapp — APPROVED** [review]

**Review by clalancette — APPROVED** [review]

**Review by ahcorde — APPROVED** [review]

**Comment by clalancette** [comment]
CI:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/12977/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/7928/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/10700/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/12954/)

**Pull request merged**